### PR TITLE
[dungeon crawler] Add replays!

### PR DESCRIPTION
**Pull request by xrigau**
Add the ability to replay games in the companion app!

### CODE REVIEW

File: dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/ArduinoLoop.java [hunk 1]
```diff
@@ -17,6 +16,7 @@
      * @param loopable the method to continuously lopp
      */
     void start(Loopable loopable) {
+        thread = new HandlerThread("Arudino Loop Mimic");
```
Comment (blundell) [inline dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/ArduinoLoop.java]: if you call start twice now, you'll leave a thread in the background thats never stopped?

File: dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/DungeonCrawlerGame.java [hunk 2]
```diff
@@ -225,12 +225,12 @@ public void stop() {
...
     @Override
     public void pause() {
-        gameEngine.pause();
+//        gameEngine.pause();
     }
 
     @Override
     public void resume() {
-        gameEngine.resume();
+//        gameEngine.resume();
```
Comment (blundell) [inline dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/DungeonCrawlerGame.java]: ?

File: dungeon-crawler/android_companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package com.novoda.dungeoncrawlercompanion
```
Comment (blundell) [inline dungeon-crawler/android_companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt]: we probably should have left the same java package name to make moving classes easier / less diff.  But not bothered now tbh sheep it :)

File: dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/EnemySpawner.java [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package com.novoda.dungeoncrawler;
 
-class EnemySpawner {
+public class EnemySpawner {
```
Comment (blundell) [inline dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/EnemySpawner.java]: hmmmmmmm

File: dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/Redux.java [hunk 5]
```diff
@@ -14,11 +14,11 @@
...
-        static final List<Particle> PARTICLE_POOL = new ArrayList<>();
-        static final List<Enemy> ENEMY_POOL = new ArrayList<>();
-        static final List<EnemySpawner> ENEMY_SPAWNER_POOL = new ArrayList<>();
-        static final List<Lava> LAVA_POOL = new ArrayList<>();
-        static final List<Conveyor> CONVEYOR_POOL = new ArrayList<>();
+        List<Particle> particlePool = new ArrayList<>();
+        List<Enemy> enemyPool = new ArrayList<>();
+        List<EnemySpawner> enemySpawnerPool = new ArrayList<>();
+        List<Lava> lavaPool = new ArrayList<>();
+        List<Conveyor> conveyorPool = new ArrayList<>();
```
Comment (blundell) [inline dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/Redux.java]: what its ok now? crazzzy

File: dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/EnemySpawner.java (hunk 4 above)
Comment (xrigau) [reply dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/EnemySpawner.java]: kotlin needs all these to be public 😓 should be "fixed" once it's all kotlin 😄

File: dungeon-crawler/android_companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt (hunk 3 above)
Comment (xrigau) [reply dungeon-crawler/android_companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt]: need a separate package for the app itself tho, but yeah we can move some classes to the same package name as the other modules

File: dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/DungeonCrawlerGame.java (hunk 2 above)
Comment (xrigau) [reply dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/DungeonCrawlerGame.java]: will delete

File: dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/ArduinoLoop.java (hunk 1 above)
Comment (xrigau) [reply dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/ArduinoLoop.java]: will call stop before starting 👍

File: dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/Redux.java (hunk 5 above)
Comment (xrigau) [reply dungeon-crawler/game_engine/src/main/java/com/novoda/dungeoncrawler/Redux.java]: yeaaah 🎉

**Pull request merged**